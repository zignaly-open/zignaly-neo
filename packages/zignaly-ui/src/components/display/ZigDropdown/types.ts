import { PopoverOrigin, PopoverPosition } from "@mui/material/Popover/Popover";

type Option = {
  id?: string;
  label?: string | JSX.Element;
  onClick?: () => void;
  href?: string;
  target?: "_blank";
  active?: boolean;
};

export type ZigDropdownOption = Option & {
  id?: string;
  children?: Option[];
  element?: JSX.Element;
  separator?: boolean;
};

export type ZigDropdownProps = {
  options: ZigDropdownOption[];
  component: ({ open }: { open: boolean }) => JSX.Element;
  anchorOrigin?: PopoverOrigin;
  anchorPosition?: PopoverPosition;
  transformOrigin?: PopoverOrigin;
};

export type ZigDropdownHandle = {
  closeDropDown?: () => void;
};
