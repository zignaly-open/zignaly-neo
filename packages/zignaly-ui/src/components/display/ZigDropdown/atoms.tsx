import React, { FC, ReactNode, useState } from "react";
import {
  ArrowBottomIconStyled,
  ChildContainer,
  ComponentSeparator,
  ComponentWrapper,
  NavLink,
  NavList,
  SpaceTaker,
  SubNavList,
  ZigDropdownContainer,
  ZigMenuItem,
} from "./styles";
import { ZigDropdownOption } from "./types";
import { MenuItem, MenuList } from "@mui/material";

export const MenuOption = ({
  option,
  index,
  onClick,
  subDropdown,
  setChildDropdownShown,
  keyName: key,
}: {
  option: ZigDropdownOption;
}) => {
  const { id, onClick } = option;
  const renderComponent = () => {
    if (option.element)
      return (
        <ZigMenuItem
          id={id}
          onClick={onClick}
          id={option.id || `dropdown-element-${index}`}
          // as={option.as}
          preserveStyles={option.preserveStyles ?? false}
          sx={{
            ...option.sx,
          }}
        >
          {option.element}
        </ZigMenuItem>
      );

    if (option.href || option.onClick) {
      return (
        <ZigMenuItem
          component={"a"}
          href={option.href}
          // onClick={option.onClick && onClick(option.onClick)}
          sx={option.sx}
          target={option?.target}
          active={option?.active}
        >
          {option.label}
        </ZigMenuItem>
      );
    }

    if (option.children) {
      return (
        <ChildContainer active={subDropdown === option}>
          <NavLink
            active={option?.active}
            id={option.id}
            customStyle={option.customStyle}
            notClickable={!option.children?.length}
            onClick={() =>
              option.children?.length && setChildDropdownShown((v) => (v ? null : option))
            }
          >
            {option.label}
            <SpaceTaker />

            {!!option.children?.length && (
              <ArrowBottomIconStyled
                style={{ color: "neutral300" }}
                rotated={subDropdown === option}
                width={"10.5px"}
              />
            )}
          </NavLink>
          {subDropdown === option && (
            <SubNavList>
              {option.children.map((c, index) => (
                <NavLink
                  id={c.id}
                  active={c?.active || undefined}
                  key={
                    "--sub-" +
                    key +
                    "--" +
                    (typeof c.label === "string" ? c.label : `-child-${index}`)
                  }
                  onClick={onClick(c.onClick!)}
                >
                  {c.label}
                </NavLink>
              ))}
            </SubNavList>
          )}
        </ChildContainer>
      );
    }
    return null;
  };

  return (
    <>
      {option.separator && <ComponentSeparator separator={true} sx={option.sx} />}
      {renderComponent()}
    </>
  );
};

export const MenuContent = ({
  id = "",
  options,
  subDropdown,
  setChildDropdownShown,
  onClick,
  matchAnchorWidth,
}) => {
  console.log("MenuContent -> options", options);
  const handleClose = () => {};

  return (
    <MenuList
      autoFocusItem={open}
      id="composition-menu"
      aria-labelledby="composition-button"
      // onKeyDown={handleListKeyDown}
      sx={{
        ".MuiMenuItem-root": {
          whiteSpace: matchAnchorWidth ? "normal" : "nowrap",
        },
      }}
    >
      {options.map((option, i) => {
        if (subDropdown && options.indexOf(subDropdown) < i) return null;
        const key =
          (option && "label" in option && typeof option.label === "string" && option.label) ||
          option.id ||
          Math.random().toString();
        return (
          <MenuOption
            option={option}
            index={i}
            key={key}
            keyName={key}
            onClick={onClick}
            subDropdown={subDropdown}
            setChildDropdownShown={setChildDropdownShown}
          />
        );
      })}
    </MenuList>
  );
};
