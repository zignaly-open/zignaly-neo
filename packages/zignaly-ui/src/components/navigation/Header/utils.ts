// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
