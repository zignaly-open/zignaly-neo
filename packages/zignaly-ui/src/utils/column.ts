import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
