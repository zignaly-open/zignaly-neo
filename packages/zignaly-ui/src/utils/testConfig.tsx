import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
