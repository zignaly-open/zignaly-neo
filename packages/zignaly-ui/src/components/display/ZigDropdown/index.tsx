import Popover from "@mui/material/Popover";
import React, { MouseEvent, useEffect, useImperativeHandle } from "react";
import { ZigDropdownProps, ZigDropdownHandle, ZigDropdownOption } from "./types";
import {
  ArrowBottomIconStyled,
  ChildContainer,
  Component,
  ComponentWrapper,
  ZigDropdownContainer,
  NavLink,
  NavList,
  SpaceTaker,
  ComponentSeparator,
  SubNavList,
} from "./styles";
import {
  Theme,
  Popper,
  ClickAwayListener,
  Grow,
  Paper,
  Button,
  MenuItem,
  MenuList,
  Box,
} from "@mui/material";
import { DropdownContent, MenuContent } from "./atoms";

const getPlacementStyles = (placement) => {
  switch (placement) {
    case "bottom-start":
      return {
        transformOrigin: "left top",
        borderRadius: "10px 10px 0 10px",
      };
    case "bottom-end":
      return {
        transformOrigin: "right top",
        borderRadius: "10px 10px 0 10px",
      };
    case "top-start":
      return {
        transformOrigin: "right bottom",
        borderRadius: "10px 0 10px 10px",
      };
    case "top-end":
    default:
      return {
        transformOrigin: "left bottom",
        borderRadius: "10px 10px 10px 0",
      };
  }
};

const ZigDropdown: (
  props: ZigDropdownProps,
  innerRef: React.Ref<ZigDropdownHandle>,
) => JSX.Element = (
  {
    id = "",
    component,
    options,
    position = "right",
    placement = "top-end",
    anchorOrigin = {
      vertical: "bottom",
      horizontal: position,
    },
    anchorPosition,
    transformOrigin = {
      vertical: "top",
      horizontal: position,
    },
    disabled,
    matchAnchorWidth = false,
    menuSx,
  }: ZigDropdownProps,
  innerRef: React.Ref<ZigDropdownHandle>,
) => {
  const [open, setOpen] = React.useState(false);
  const anchorRef = React.useRef<HTMLButtonElement>(null);

  const handleToggle = () => {
    setOpen((prevOpen) => !prevOpen);
  };

  // const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const [childDropdownShow, setChildDropdownShown] = React.useState<null | ZigDropdownOption>(null);
  // const isOpen = !!anchorRef.current;
  // const handleToggle = (event: React.MouseEvent<HTMLElement>) =>
  //   !disabled ? setAnchorEl((v) => (v ? null : event.currentTarget)) : () => {};

  // const handleClose = () => {
  //   setChildDropdownShown(null);
  //   setAnchorEl(null);
  // };

  const handleClose = (event: Event | React.SyntheticEvent) => {
    if (anchorRef.current && anchorRef.current.contains(event.target as HTMLElement)) {
      return;
    }

    setOpen(false);
  };

  const placementStyles = getPlacementStyles(placement);
  console.log(placementStyles);

  // function handleListKeyDown(event: React.KeyboardEvent) {
  //   if (event.key === "Tab") {
  //     event.preventDefault();
  //     setOpen(false);
  //   } else if (event.key === "Escape") {
  //     setOpen(false);
  //   }
  // }

  // // return focus to the button when we transitioned from !open -> open
  // const prevOpen = React.useRef(open);
  // React.useEffect(() => {
  //   if (prevOpen.current === true && open === false) {
  //     anchorRef.current!.focus();
  //   }

  //   prevOpen.current = open;
  // }, [open]);

  // useImperativeHandle(
  //   innerRef,
  //   () => ({
  //     closeDropDown: () => {
  //       handleClose();
  //     },
  //     open,
  //   }),
  //   [anchorEl, open],
  // );
  const onClick = (f: () => void) => (e: MouseEvent<HTMLAnchorElement>) => {
    e.preventDefault();
    handleClose();
    // the timeout here is needed because an action can trigger Suspense and then the popover will remain somewhere on the page
    setTimeout(() => {
      if (e.button === 1) {
        // middle button fix for Safari
        window.open(e.currentTarget.href);
      } else {
        f();
      }
    });
  };

  // useEffect(() => {
  //   window.addEventListener("scroll", handleClose);
  //   return () => {
  //     window.removeEventListener("scroll", handleClose);
  //   };
  // }, []);
  return (
    <>
      {/* <Button
        ref={anchorRef}
        id="composition-button"
        aria-controls={open ? "composition-menu" : undefined}
        aria-expanded={open ? "true" : undefined}
        aria-haspopup="true"
        onClick={handleToggle}
      >
        Dashboard
      </Button> */}
      <Box ref={anchorRef} role="button" onClick={handleToggle} id={id && `${id}-container`}>
        {component({ open })}
      </Box>
      {options && (
        // <Popover
        //   id="popover-menu"
        //   anchorEl={anchorEl}
        //   open={open}
        //   disableScrollLock={true}
        //   slotProps={{
        //     paper: {
        //       sx: {
        //         width: matchAnchorWidth ? anchorEl?.offsetWidth : "auto",
        //         minWidth: "220px",
        //         backgroundColor: "neutral800",
        //         color: "#fff",
        //         boxShadow: "0 4px 6px -2px #00000061",
        //         borderRadius: `${position === "right" ? 4 : 0}px ${
        //           position === "right" ? 0 : 4
        //         } 4px 4px`,
        //         border: "none",
        //         ...menuSx,
        //       },
        //     },
        //   }}
        //   onClose={handleClose}
        //   anchorPosition={anchorPosition}
        //   transformOrigin={transformOrigin}
        //   anchorOrigin={anchorOrigin}
        // >

        <Popper
          open={open}
          anchorEl={anchorRef.current}
          // anchorEl={anchorEl}
          role={undefined}
          placement={placement}
          transition
          // disablePortal
          sx={{
            minWidth: matchAnchorWidth ? anchorRef.current?.offsetWidth : "220px",
            zIndex: 1300,
            // color: "#fff",
            " > div": {
              boxShadow: "0 4px 6px -2px #00000061",
              borderRadius: placementStyles.borderRadius,
              // borderRadius: "4px",
              overflow: "hidden",
              backgroundColor: "neutral800",
            },
            ".MuiList-root": {},
            ...menuSx,
          }}
        >
          {({ TransitionProps, placement }) => (
            <Grow
              {...TransitionProps}
              style={{
                transformOrigin: placementStyles.transformOrigin,
              }}
              timeout={open ? 180 : 0}
            >
              <div>
                <ClickAwayListener onClickAway={handleClose}>
                  <div>
                    <MenuContent
                      id={id}
                      options={options}
                      setChildDropdownShown={setChildDropdownShown}
                      subDropdown={childDropdownShow}
                      onClick={onClick}
                      matchAnchorWidth={matchAnchorWidth}
                    />
                  </div>
                </ClickAwayListener>
              </div>
            </Grow>
          )}
        </Popper>
      )}
    </>
  );
};

export default React.forwardRef(ZigDropdown);

export type {
  ZigDropdownProps,
  ZigDropdownHandle as ZigDropdownHandleType,
  ZigDropdownOption as ZigDropdownOptionType,
} from "./types";
