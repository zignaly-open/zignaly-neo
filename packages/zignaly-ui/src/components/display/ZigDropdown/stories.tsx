import React, { useRef, useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import ZigDropdown, { ZigDropdownHandleType } from "./index";
import { Box } from "@mui/material";
import ZigButton from "../../inputs/ZigButton";
import EditIcon from "@mui/icons-material/Edit";

// type ZigDropdownProps = Partial<React.ComponentProps<typeof ZigDropdown>>;
type ZigDropdownProps = React.ComponentProps<typeof ZigDropdown>;
const ZigDropdownWrapper = (props: ZigDropdownProps) => {
  console.log(props);
  const ref = useRef<ZigDropdownHandleType>(null);
  return (
    <ZigDropdown
      ref={ref}
      component={(open) => (
        <ZigButton narrow variant={"outlined"} active={!!open}>
          <EditIcon />
        </ZigButton>
      )}
      options={[
        {
          label: "Important link 1",
          id: "dropwown-one",
          target: "_blank",
          href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
        },
        {
          label: "Important link 2",
          id: "dropwown-one",
          target: "_blank",
          href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
        },
        { separator: true },
        {
          label: "Important link 3 with separator",
          id: "dropwown-one",
          target: "_blank",
          href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
        },
        {
          label: "Alert",
          id: "dropwown-one",
          target: "_blank",
          onClick: () => alert(),
        },
        {
          label: "Close from inside",
          id: "dropwown-one",
          target: "_blank",
          onClick: () => ref.current?.closeDropDown?.(),
        },
        {
          element: <Box sx={{ background: "#f00" }}>Element</Box>,
        },
      ]}
      {...props}
    />
  );
};

// type Story = StoryObj<ZigDropdownProps>;
const meta = {
  title: "Display/ZigDropdown",
  component: ZigDropdown,
  render: (props) => <ZigDropdownWrapper {...props} />,
  argTypes: {
    component: {
      description: "(open: boolean) => JSX.Element",
    },
    options: {
      description: "ZigDropdownOption[]",
    },
    placement: {
      control: "select",
      options: [
        ...["top", "bottom", "right", "left", "start", "end", "auto"],
        ...[
          "top-start",
          "top-end",
          "bottom-start",
          "bottom-end",
          "right-start",
          "right-end",
          "left-start",
          "left-end",
        ],
        ...["auto", "auto-start", "auto-end"],
      ],
      description: "Menu popper placement",
    },
    matchAnchorWidth: {
      control: "boolean",
      description: "Match anchor width",
    },
  },
  args: {
    placement: "bottom-start",
  },
} as Meta<ZigDropdownProps>;
export default meta;
// type ZigDropdownProps = React.ComponentProps<typeof ZigDropdown>;
type Story = StoryObj<ZigDropdownProps>;

export const Default: Story = {};

// export const Default: Story;

// const DropdownStorybookCrutch: React.FC = () => {
//   const ref = useRef<ZigDropdownHandleType>(null);
//   return (

//   );
// };

// export default {
//   title: "Display/ZigDropdown",
//   component: DropdownStorybookCrutch,
//   argTypes: {
//     component: {
//       description: "(open: boolean) => JSX.Element",
//     },
//     options: {
//       description: "ZigDropdownOption[]",
//     },
//   },
// } as ComponentMeta<typeof DropdownStorybookCrutch>;

// const Template: ComponentStory<typeof DropdownStorybookCrutch> = (args) => {
//   return <DropdownStorybookCrutch {...args} />;
// };

// export const Basic = Template.bind({});
// Basic.args = {};
