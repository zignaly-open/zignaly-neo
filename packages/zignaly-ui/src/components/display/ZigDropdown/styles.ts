import styled, { css } from "styled-components";
import { styledIf } from "utils/styled";
import { ReactComponent as ArrowBottomIcon } from "assets/icons/caret-down.svg";
import dark from "theme/dark";

const withSeparator = (props: WithSeparator) =>
  props.separator &&
  css`
    border-top: 1px dotted ${({ theme }) => theme.neutral600};
    margin: 6px 35px 6px;
  `;

type WithSeparator = { separator?: boolean };
type WithCustomStyle = { customStyle?: string };

export const ZigDropdownContainer = styled.div`
  user-select: none;
  min-width: 245px;
`;

export const Component = styled.div`
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
`;

export const ComponentWrapper = styled.div<WithSeparator & WithCustomStyle>`
  padding: 6px 32px;

  ${(props) => props.customStyle || ""};
  ${withSeparator}
`;

export const NavLink = styled.span<
  {
    notClickable?: boolean;
    active?: boolean;
    disabled?: boolean;
    customStyle?: string;
  } & WithSeparator &
    WithCustomStyle
>`
  color: ${dark.neutral200};
  font-weight: 400;
  font-size: 14px;
  line-height: 28px;
  padding: 6px 32px;
  letter-spacing: 0.55px;
  text-decoration: none;
  transition: 0.15s linear;

  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  overflow: hidden;
  max-width: 245px;

  svg {
    transition: 0.15s linear;
  }

  ${({ notClickable }) =>
    !notClickable &&
    css`
      &:hover {
        background: rgb(28, 29, 53);
      }
    `}

  ${({ active }) => `
    ${styledIf(
      active,
      `
      color: #7682f7;
    `,
      `

      &:not([disabled]) {
        cursor: pointer;

        &:hover {
          color: #fff;
        }
      }
    `,
    )}
  `}

  ${(props) =>
    props.notClickable &&
    css`
      cursor: default !important;
    `}

  ${(props) => props.customStyle || ""};
  ${withSeparator};
`;

export const ComponentSeparator = styled.div<
  WithSeparator & {
    customStyle?: string;
  }
>`
  ${withSeparator};
  ${(props) => props.customStyle || ""};
`;

export const NavList = styled.div`
  display: flex;
  flex-direction: column;
  > *:first-child {
    margin-top: 11px;
  }
  > *:last-child {
    margin-bottom: 6px;
  }
`;

export const SubNavList = styled.div`
  display: flex;
  flex-direction: column;
  > *:first-child {
    margin-top: 6px;
  }
  > *:last-child {
    margin-bottom: 6px;
  }
`;

export const ChildContainer = styled.div<{ active: boolean } & WithSeparator>`
  ${(props) =>
    props.active &&
    css`
      background: rgb(25, 26, 48);
      margin-bottom: 0 !important;
    `}

  ${withSeparator}
`;

export const SpaceTaker = styled.span`
  display: flex;
  flex: 1;
`;

export const ArrowBottomIconStyled = styled(ArrowBottomIcon)<{ rotated?: boolean }>`
  transition: 0.15s linear;
  transform: rotate(${(props) => (props.rotated ? "180deg" : "0deg")});
`;
