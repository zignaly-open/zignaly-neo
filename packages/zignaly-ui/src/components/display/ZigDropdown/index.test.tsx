import React from "react";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, waitFor, screen } from "@testing-library/react";
import ZigDropdown from ".";

describe("components/ZigDropdown", () => {
  const options = [
    { id: "option1", label: "Option 1", onClick: jest.fn() },
    { id: "option2", label: "Option 2", onClick: jest.fn() },
  ];

  const DropdownComponent = () => <div>Component</div>;

  it("should render without crashing", () => {
    const { container } = renderWithProvidersUi(
      <ZigDropdown id="test-id" component={DropdownComponent} options={options} />,
    );

    const dropdown = container.querySelector("#test-id-container") as HTMLElement;
    expect(dropdown).toBeVisible();
    expect(dropdown).toMatchSnapshot();
  });

  it("should open dropdown on click", () => {
    renderWithProvidersUi(
      <ZigDropdown id="test-id" component={DropdownComponent} options={options} />,
    );

    const dropdown = screen.getByRole("button");
    fireEvent.click(dropdown);

    const listItem = screen.getByText(options[0].label);
    expect(listItem).toBeVisible();
  });

  it("should trigger onClick for the option when option is clicked", async () => {
    renderWithProvidersUi(
      <ZigDropdown id="test-id" component={DropdownComponent} options={options} />,
    );

    const dropdown = screen.getByRole("button");
    fireEvent.click(dropdown);

    const listItem = screen.getByText(options[0].label);
    fireEvent.click(listItem);

    await waitFor(() => {
      expect(options[0].onClick).toHaveBeenCalled();
    });
  });

  it("should display component", () => {
    renderWithProvidersUi(
      <ZigDropdown id="test-id" component={DropdownComponent} options={options} />,
    );

    const component = screen.getByText("Component");
    expect(component).toBeVisible();
  });
});
