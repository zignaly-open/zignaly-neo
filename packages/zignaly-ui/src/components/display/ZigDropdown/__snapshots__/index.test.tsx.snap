// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ZigDropdown should render without crashing 1`] = `
<div
  class="sc-hKMtZM eCiABY"
  id="test-id-container"
  role="button"
>
  <div>
    Component
  </div>
</div>
`;
